src/main/java/io/github/zap/commons/BaseZapPlugin.java

```diff
@@ -0,0 +1,45 @@
+package io.github.zap.commons;
+
+import com.google.inject.Injector;
+import org.apache.commons.lang3.time.StopWatch;
+import org.bukkit.plugin.java.JavaPlugin;
+
+/**
+ * A base plugin for *all* zap plugin projects
+ */
+public abstract class BaseZapPlugin extends JavaPlugin {
```

Comment:
IMO this should also have overrides for onDisable() and onLoad(), along with more generalized code for timing so we could reuse the same code for enable, disable, load

Reply:
Currently idk what to add about `onDisable` and `onLoad` maybe another timer?

Reply:
Yeah I think they just should have a timer. You can probably make a private method to avoid duplicating code